metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py

```diff
@@ -0,0 +1,341 @@
+            client_id=self.__config.client_id,
+            client_secret=self.__config.client_secret,
+            tenant_id=self.__config.tenant_id,
+        )
+
+        self.__admin_api_resolver = AdminAPIResolver(
+            client_id=self.__config.client_id,
+            client_secret=self.__config.client_secret,
+            tenant_id=self.__config.tenant_id,
+        )
+
+    def _get_dashboard_endorsements(
+        self, scan_result: Optional[dict]
+    ) -> Dict[str, List[str]]:
+        """
+        Store saved dashboard endorsements into a dict with dashboard id as key and
+        endorsements or tags as list of strings
+        """
+        results: Dict[str, List[str]] = {}
+        if scan_result is None:
+            return results
+
+        for scanned_dashboard in scan_result["dashboards"]:
+            # Iterate through response and create a list of PowerBiAPI.Dashboard
+            dashboard_id = scanned_dashboard.get("id")
+            tags = self._parse_endorsement(
+                scanned_dashboard.get("endorsementDetails", None)
+            )
+            results[dashboard_id] = tags
+
+        return results
+
+    def _get_report_endorsements(
+        self, scan_result: Optional[dict]
+    ) -> Dict[str, List[str]]:
+        results: Dict[str, List[str]] = {}
+
+        if scan_result is None:
+            return results
+
+        reports: List[dict] = scan_result.get("reports", [])
+
+        for report in reports:
+            report_id = report.get("id", "")
+            endorsements = self._parse_endorsement(
+                report.get("endorsementDetails", None)
+            )
+            results[report_id] = endorsements
+
+        return results
+
+    def _get_resolver(self):
+        if self.__config.admin_apis_only:
+            return self.__admin_api_resolver
+        return self.__regular_api_resolver
+
+    def _get_entity_users(
+        self, workspace_id: str, entity_name: str, entity_id: str
+    ) -> List[User]:
+        """
+        Return list of dashboard users
+        """
+        users: List[User] = []
+        if self.__config.extract_ownership is False:
+            logger.info(
+                "Extract ownership capabilities is disabled from configuration and hence returning empty users list"
+            )
+            return users
+
+        try:
+            users = self.__admin_api_resolver.get_users(
+                workspace_id=workspace_id,
+                entity=entity_name,
+                entity_id=entity_id,
+            )
+        except requests.exceptions.HTTPError as e:
+            if data_resolver.is_permission_error(e):
+                logger.warning(
+                    f"{entity_name} users would not get ingested as admin permission is not enabled on "
+                    "configured Azure AD Application",
+                )
+                return users
+            # if Other error then re-raise
+            raise e
```

Comment:
Why are we raising here? is failing to get_users for another reason meaning we should fail the entire connector? Why?

Reply:
Will a warning do okay here?

Reply:
Yup ok